### PR TITLE
Two methods added to UserApi to upload and delete profile image.

### PR DESCRIPTION
**Pull request by floor12**
I need it in my project, so i added this methods to update profiles pictures.

### CODE REVIEW

**Comment by floor12** [comment]
Hello, man! Do you have some time to watch my pull-request? )

File: src/Api/UsersApi.php [hunk 1]
```diff
@@ -319,4 +320,50 @@ public function getUserStatus(string $userId)
 
         return $this->handleResponse($response, UserStatus::class);
     }
+
```
Comment (thePanz) [inline src/Api/UsersApi.php:323]: Remove the extra empty line

File: src/Api/UsersApi.php [hunk 2]
```diff
@@ -319,4 +320,50 @@ public function getUserStatus(string $userId)
...
+    {
+        if (empty($userId)) {
+            throw new InvalidArgumentException('UserId can not be empty');
+        }
+
+        $response = $this->httpDelete(sprintf('/users/%s/image', $userId));
+        return $this->handleResponse($response, Status::class);
+    }
+
+
```
Comment (thePanz) [inline src/Api/UsersApi.php]: Remove the extra empty line

**Review by thePanz — CHANGES_REQUESTED** [review]

**Comment by thePanz** [comment]
Thank you @floor12 for your MR!
I left a couple of minor comments, LGTM otherwise.

Tests are missing, would you be able to add them?

**Comment by floor12** [comment]
I usually do tests right away. But this time for some reason I did not immediately figure it out with them. I will add tests in `Pnz\MattermostClient\Tests\Api\UsersTest` soon.

**Comment by floor12** [comment]
I added some tests, but, I dint find how I can check row post request. I also found that you did not test the same case with file posting in `FilesApi::sendFile`.

**Comment by floor12** [comment]
Hey man!)

File: tests/Api/UsersTest.php [hunk 3]
```diff
@@ -83,7 +83,7 @@ public function testLoginEmptyPassword(): void
     public function testGetUserByEmailSuccess(): void
     {
         $userEmail = 'user@example.com';
-        $this->configureMessage('GET', '/users/email/'.$userEmail);
+        $this->configureMessage('GET', '/users/email/' . $userEmail);
```
Comment (thePanz) [inline tests/Api/UsersTest.php]: why those changes? :thinking:

File: tests/Api/UsersTest.php [hunk 4]
```diff
@@ -517,4 +517,33 @@ public function testGetChannelPostsException(string $exception, int $code): void
...
+    public function testDeleteProfileImageEmptyId(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->client->deleteProfileImage('');
+    }
+
+    public function testDeleteProfileImageSuccess(): void
+    {
+        $userId = '1234';
+        $this->configureMessage('DELETE', "/users/{$userId}/image");
```
Comment (thePanz) [inline tests/Api/UsersTest.php]: please use single quotes and concatenation (as in the other tests)

File: tests/Api/UsersTest.php (hunk 3 above)
Comment (floor12) [reply tests/Api/UsersTest.php]: It was was auto formatted by default php code format settings in phpStorm.

**Comment by floor12** [comment]
I updated quotes. And concatenation update was reverted few commits before. It was made by mistake. Latest version doesn't include concatenation style updates, only 2 new methods and tests.

**Comment by floor12** [comment]
Hey man.

**Review by thePanz — APPROVED** [review]

**Comment by thePanz** [comment]
Thanks @floor12! I applied some additional changes and merged in 8c42c69

Tthe $image parameter can be a string, a resource or a StreamInterface instance.

**Pull request closed without merging**